include/deal.II/dofs/dof_renumbering.h

```diff
@@ -689,6 +690,76 @@ namespace DoFRenumbering
                  const std::vector<unsigned int> &target_component =
                    std::vector<unsigned int>());
 
+
+
+  /**
+   * Convenience alias for variant type of supported FEValuesExtractors structs
+   */
+  using ExtractorVariant = std::variant<FEValuesExtractors::Scalar,
+                                        FEValuesExtractors::Vector,
+                                        FEValuesExtractors::Tensor<2>,
+                                        FEValuesExtractors::SymmetricTensor<2>>;
```

Comment:
The use of a `std::variant` here is clever -- I did not think of this in other places. We may want to encourage this in other places. I think this could perhaps move into the header files where we declare the extractors.